nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -189,7 +189,7 @@ def _parse_args(self, arg_names: List, args: Tuple, kwargs: Dict, def_kwargs: Di
                 raise TypeError(f"{self.__class__.__name__}.__init__() got multiple values for argument '{key}'")
             merged_kwargs[key] = value
         for key, value in def_kwargs.items():
-            if key not in merged_kwargs:
+            if key not in merged_kwargs and key in arg_names:
```

Comment:
I don't really want to know all the details here, but I still suggest writing a full "bug fixes + enhancements" list in the PR description, so that I have materials to write in the release note.

Reply:
add them